### PR TITLE
feat(notifier): adds source to notifier channels

### PR DESCRIPTION
**Pull request by jurajpiar**
Resolves https://rsklabs.atlassian.net/browse/RMKT-849

Plus refactors the channels table (out), adding the info to plan model.
Also removes all notifier migrations as the notifier feature is not in production yet.

Dependency of https://github.com/rsksmart/rif-marketplace-ui/pull/942

### CODE REVIEW

File: src/services/notifier/api/notifierSvcProvider.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+}
+
+export type ChannelMapper = (value: NotificationServiceTypeDTO, index: number, array: NotificationServiceTypeDTO[]) => NotifierChannel
+
+const ERROR_PFX = 'Notifier failed at endpoint'
+
+const RESOURCES = {
+  getSubscriptions: 'getSubscriptions',
+  getSubscriptionPlans: 'getSubscriptionPlans',
+  getAvailableChannels: 'getAvailableChannels'
```
Comment (vjgene) [inline src/services/notifier/api/notifierSvcProvider.ts]: the endpoint is /info/availableNotificationPreferences

Comment (jurajpiar) [reply src/services/notifier/api/notifierSvcProvider.ts]: ok

File: src/migrations/scripts/20210616061834-notifiersignature.ts [hunk 2]
```diff
@@ -1,11 +0,0 @@
-import Sequelize, { QueryInterface } from 'sequelize'
-
-export default {
-  async up (queryInterface: QueryInterface): Promise<void> {
-    await queryInterface.addColumn('notifier_subscription', 'signature', Sequelize.STRING)
-  },
-
-  async down (queryInterface: QueryInterface): Promise<void> {
-    await queryInterface.removeColumn('notifier_subscription', 'signature')
```
Comment (0xartem) [inline src/migrations/scripts/20210616061834-notifiersignature.ts:9]: why is this removed?

File: src/migrations/scripts/20210728154020-notifierWithdraw.ts [hunk 3]
```diff
@@ -1,11 +0,0 @@
-import Sequelize, { QueryInterface } from 'sequelize'
-
-export default {
-  async up (queryInterface: QueryInterface): Promise<void> {
-    await queryInterface.addColumn('notifier_subscription', 'withdrawableFunds', Sequelize.STRING)
-  },
-
-  async down (queryInterface: QueryInterface): Promise<void> {
-    await queryInterface.removeColumn('notifier_subscription', 'withdrawableFunds')
```
Comment (0xartem) [inline src/migrations/scripts/20210728154020-notifierWithdraw.ts:9]: and this?

File: src/services/notifier/hooks/plans.hook.ts [hunk 4]
```diff
@@ -61,19 +54,15 @@ export default {
...
-              as: 'channels',
-              required: true,
-              attributes: ['name'],
-              where: {
-                name: channels
+          paramsSeq.where = {
+            ...paramsSeq.where ?? {},
+            channels: {
+              [Op.contains]: {
+                type: channels
```
Comment (vjgene) [inline src/services/notifier/hooks/plans.hook.ts]: Considering we are storing origin also as part of json, wouldn't this query return non-relevant results for ex. if email address is apiuser@gmail.com or if the url is https://email.domain.com

File: src/services/notifier/hooks/subscriptions.hook.ts [hunk 5]
```diff
@@ -100,15 +99,7 @@ export default {
...
-              where: literal('plan.id = subscriptionPlanId'),
-              include: [
-                {
-                  model: NotifierChannelModel,
-                  as: 'channels',
-                  attributes: ['name'],
-                  required: true
-                }
-              ]
+              where: literal('plan.id = subscriptionPlanId')
```
Comment (vjgene) [inline src/services/notifier/hooks/subscriptions.hook.ts]: May be it's a good idea to add comment that we store the cache planid as subscriptionPlanId  in subscription, instead of the actual provider plan id

**Review by vjgene — APPROVED** [review]
Nice work! couple of minor comments

File: src/services/notifier/api/notifierSvcProvider.ts [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+export type NotifierResult<T> = {
+    message: | 'OK' // TODO: add the rest of the options
+    status: | 'OK' // TODO: add the rest of the options
+    content: T
+}
+
+export type ChannelMapper = (value: NotificationServiceTypeDTO, index: number, array: NotificationServiceTypeDTO[]) => NotifierChannel
+
+const ERROR_PFX = 'Notifier failed at endpoint'
```
Comment (itofarina) [inline src/services/notifier/api/notifierSvcProvider.ts]: If I understood correctly, this message is quite generic and doesn't seem to be related to a PFX error. Can we call it something else?

File: test/unit/services/notifier/api/notifierSvcProvider.spec.ts [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    await sequelize.sync({ force: true })
+  })
+
+  afterEach(() => {
+    sandbox.restore()
+  })
+
+  after(() => {
+    sandbox.restore()
+  })
```
Comment (itofarina) [inline test/unit/services/notifier/api/notifierSvcProvider.spec.ts:44]: Isn't `afterEach` enough?

File: src/services/notifier/hooks/plans.hook.ts [hunk 8]
```diff
@@ -61,19 +54,15 @@ export default {
...
               }
             }
-          ]
-        }
+          }
+        } // TODO: needs testing
```
Comment (itofarina) [inline src/services/notifier/hooks/plans.hook.ts]: I think it could be better to remove it then or update the comment to "not used yet and not tested"

**Review by itofarina — APPROVED** [review]
Looks great! Just minor comments

File: src/services/notifier/api/notifierSvcProvider.ts (hunk 6 above)
Comment (jurajpiar) [reply src/services/notifier/api/notifierSvcProvider.ts]: PFX is just short for prefix, but probably would be clearer as`ERROR_MSG_PFX`

File: test/unit/services/notifier/api/notifierSvcProvider.spec.ts (hunk 7 above)
Comment (jurajpiar) [reply test/unit/services/notifier/api/notifierSvcProvider.spec.ts:44]: no, seems to be needed. Otherwise it seems to leak to later test files

File: src/services/notifier/hooks/plans.hook.ts (hunk 4 above)
Comment (jurajpiar) [reply src/services/notifier/hooks/plans.hook.ts]: no this is just to filter channel types like API, EMAIL, etc

Comment (jurajpiar) [reply src/services/notifier/hooks/plans.hook.ts]: but it's also not used so I might just remove it.

File: src/services/notifier/hooks/subscriptions.hook.ts (hunk 5 above)
Comment (jurajpiar) [reply src/services/notifier/hooks/subscriptions.hook.ts]: I disagree, as it should be pretty obvious from the naming, but I will add it there.

**Review by 0xartem — APPROVED** [review]

File: src/migrations/scripts/20210616061834-notifiersignature.ts (hunk 2 above)
Comment (jurajpiar) [reply src/migrations/scripts/20210616061834-notifiersignature.ts:9]: because it is moved to notifier-init migration

File: src/migrations/scripts/20210728154020-notifierWithdraw.ts (hunk 3 above)
Comment (jurajpiar) [reply src/migrations/scripts/20210728154020-notifierWithdraw.ts:9]: because it is moved to notifier-init migration

File: src/migrations/scripts/20210616061834-notifiersignature.ts (hunk 2 above)
Comment (jurajpiar) [reply src/migrations/scripts/20210616061834-notifiersignature.ts:9]: https://github.com/rsksmart/rif-marketplace-cache/pull/787/files#diff-ccf35de765e23ca21519b8651936f8242792c7e7f9d85107997adfeeb8796994R180-R184

File: src/migrations/scripts/20210728154020-notifierWithdraw.ts (hunk 3 above)
Comment (jurajpiar) [reply src/migrations/scripts/20210728154020-notifierWithdraw.ts:9]: https://github.com/rsksmart/rif-marketplace-cache/pull/787/files#diff-ccf35de765e23ca21519b8651936f8242792c7e7f9d85107997adfeeb8796994R254
